terminal_gui.py

```diff
@@ -98,9 +99,31 @@ def get_input():
             print_input_field(stdscr, text)
         return text
 
-    curses.curs_set(2)  # Show cursor
+    curses.curs_set(1)  # Show cursor
     stdscr.keypad(True)  # Enable keypad for non-character keys
 
     text_input_value = get_input()
     curses.endwin()  # End curses window
     return text_input_value
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument(
+    '-H',
+    '--use-hugging-face',
+    dest='use_hugging_face',
+    action="store_true",
+    help='Use Hugging Face as the model provider'
+)
+USE_HUGGING_FACE = parser.parse_args().use_hugging_face
```

Comment:
there is no need to introduce another new variable, you can use `USE_OLLAMA` and set it to false when `-H` flag is present.

```suggestion
USE_OLLAMA = not parser.parse_args().use_hugging_face
```